qiskit/utils/deprecation.py

```diff
@@ -353,8 +355,13 @@ def _write_deprecation_msg(
     since: str,
     pending: bool,
     additional_msg: str,
-    removal_timeline: str,
+    removal_timeline: str | None,
 ) -> tuple[str, Type[DeprecationWarning] | Type[PendingDeprecationWarning]]:
+    if not removal_timeline:
+        # TODO: remove the cast `str()` once https://github.com/Qiskit/qiskit-aer/pull/2006 is released
+        removal_major = int(str(since).split(".")[0]) + 1
```

Comment:
```suggestion
        # TODO: remove the cast `str()` once https://github.com/Qiskit/qiskit-aer/pull/2006 is released
        removal_major = int(since.split(".")[0]) + 1
```